pkgs/tools/archivers/7zz/default.nix

```diff
@@ -110,18 +108,18 @@ stdenv.mkDerivation rec {
     };
   };
 
-  meta = with lib; {
```

Comment:
this with is simple and common

Reply:
"common" is irrelevant, and it's unnecessary.

Reply:
I did not double-check cross compilation, will do so now.

Reply:
common is **completely** relevant to maintaining a codebase with fellow contributors

And what do you mean unnecessary. Removing this with requires 4 references to `lib.` when it could just be left as it is.

If having to write out 4 copies of `lib.` is fine why is `with` allowed for `maintainers` when it's "unnecessary" you can just do

`maintainers = [ lib.maintainers.anna328p lib.maintainers.eclairevoyant lib.maintainers.jk lib.maintainers.peterhoeg ];`


The downside to `with` are that there's no clarity as to where the value came from when there are multiple, but that's the only `with` up to that point so completely fine

Reply:
common is completely irrelevant to improving code quality. and no committer has complained about this change in the past.

Reply:
Honestly, every reviewer seems to have their own ideas of what is acceptable, so I'm not interested on bikeshedding a cosmetic change. If you have feedback on the actual build process etc I'll discuss that.

Reply:
please revert the cosmetic change then

Reply:
I'd prefer if there is some unified documentation about best practices for nixpkgs. If you can point to that, I will gladly make the change.

Reply:
Alright, as https://nixos.org/manual/nixpkgs/stable/#chap-meta uses `with lib;` I will stick with that. Let the record show that I'm not interested in un-reverting, then.

Reply:
As I mentioned it's the common pattern across nixpkgs

```
λ rg "meta = with lib; \{" . --stats -q | rg -e "\d+ matched lines"
21126 matched lines

λ rg "meta = \{" . --stats -q | rg -e "\d+ matched lines"
12683 matched lines

# There are a couple files you may want to filter out of the with lib search but very few
λ rg "meta = with lib; \{" . --count | sed 's/:/ /g' | sort -nr -k2 | head -n20
./pkgs/servers/http/nginx/modules.nix 49
./pkgs/applications/editors/eclipse/plugins.nix 29
./pkgs/applications/misc/octoprint/plugins.nix 22
./pkgs/development/libraries/hunspell/dictionaries.nix 13
./pkgs/applications/editors/jetbrains/default.nix 12
./pkgs/servers/search/elasticsearch/plugins.nix 9
./pkgs/misc/tmux-plugins/default.nix 7
./doc/languages-frameworks/python.section.md 7
./pkgs/development/tools/continuous-integration/buildbot/plugins.nix 6
./pkgs/development/python-modules/textx/default.nix 6
./pkgs/development/libraries/unixODBCDrivers/default.nix 6
./pkgs/applications/video/vdr/plugins.nix 6
./pkgs/applications/networking/znc/modules.nix 6
./pkgs/applications/office/activitywatch/default.nix 5
./pkgs/top-level/lua-packages.nix 4
./pkgs/development/tools/replay-io/default.nix 4
./pkgs/data/fonts/noto-fonts/default.nix 4
./pkgs/build-support/setup-hooks/patch-rc-path-hooks/default.nix 4
./pkgs/applications/editors/kakoune/plugins/overrides.nix 4
./pkgs/top-level/perl-packages.nix 3

# For raw meta there are many generated files
λ rg "meta = \{" . --count | sed 's/:/ /g' | sort -nr -k2 | head -n20
./pkgs/development/lisp-modules/imported.nix 4598
./pkgs/top-level/perl-packages.nix 1827
./pkgs/development/lisp-modules-new-obsolete/imported.nix 856
./pkgs/applications/editors/emacs/elisp-packages/elpa-devel-generated.nix 406
./pkgs/development/node-packages/node-packages.nix 396
./pkgs/applications/editors/emacs/elisp-packages/elpa-generated.nix 396
./pkgs/applications/editors/vscode/extensions/default.nix 244
./pkgs/applications/editors/emacs/elisp-packages/nongnu-generated.nix 211
./pkgs/development/lua-modules/generated-packages.nix 111
./pkgs/applications/emulators/retroarch/cores.nix 90
./pkgs/tools/misc/graylog/plugins.nix 20
./pkgs/development/compilers/elm/packages/node-packages.nix 15
./pkgs/applications/science/electronics/bitscope/packages.nix 8
./pkgs/misc/tmux-plugins/default.nix 7
./pkgs/development/python-modules/fenics/default.nix 6
./pkgs/development/ocaml-modules/janestreet/old.nix 6
./pkgs/applications/editors/vim/plugins/overrides.nix 6
./pkgs/applications/networking/browsers/firefox/packages.nix 5
./pkgs/top-level/release-haskell.nix 4
./pkgs/development/libraries/aspell/dictionaries.nix 4

# This is the effect of the filter
λ rg "meta = \{" . -g '!*generated*' -g '!*lisp-modules*' -g '!*perl-packages.nix' -g '!*node-packages.nix' -g '!*vscode' --count  | sed 's/:/ /g' | sort -nr -k2 | head -n20
./pkgs/applications/emulators/retroarch/cores.nix 90
./pkgs/tools/misc/graylog/plugins.nix 20
./pkgs/applications/science/electronics/bitscope/packages.nix 8
./pkgs/misc/tmux-plugins/default.nix 7
./pkgs/development/python-modules/fenics/default.nix 6
./pkgs/development/ocaml-modules/janestreet/old.nix 6
./pkgs/applications/editors/vim/plugins/overrides.nix 6
./pkgs/applications/networking/browsers/firefox/packages.nix 5
./pkgs/top-level/release-haskell.nix 4
./pkgs/development/libraries/aspell/dictionaries.nix 4
./pkgs/build-support/setup-hooks/patch-rc-path-hooks/test/default.nix 4
./pkgs/os-specific/linux/g15daemon/default.nix 3
./pkgs/development/interpreters/bats/libraries.nix 3
./pkgs/applications/virtualization/xen/packages.nix 3
./pkgs/applications/networking/znc/modules.nix 3
./pkgs/applications/graphics/gimp/plugins/default.nix 3
./pkgs/top-level/haxe-packages.nix 2
./pkgs/tools/nix/nixos-render-docs/src/tests/test_lists.py 2
./pkgs/stdenv/linux/make-bootstrap-tools.nix 2
./pkgs/servers/search/manticoresearch/default.nix 2

# and these are the filtered results
λ rg "meta = \{" . -g '!*generated*' -g '!*lisp-modules*' -g '!*perl-packages.nix' -g '!*node-packages.nix' -g '!*vscode' --stats -q | rg -e "\d+ matched lines"
3597 matched lines

# even with that filter and a bunch of extra bits that likely trim out plenty of non-generated results `with lib;` still has it's huge amount of matches
λ rg "meta = with lib; \{" . -g '!*generated*' -g '!*lisp-modules*' -g '!*perl-packages.nix' -g '!*node-packages.nix' -g '!*vscode' -g '!*nginx' -g '!*eclipse' -g '!*octoprint' -g '!*hunspell' -g '!*jetbrains' -g '!*elasticsearch' --stats -q | rg -e "\d+ matched lines"
20974 matched lines

# This is still almost 6 times as many matches
```

There's also this example in the nixpkgs documentation for how to write a meta

https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#meta-attributes-chap-meta

---

A maintainer commenting on a change should not be responded with scare quotes and dismissing the comment immediately as irrelevant.

You also shouldn't bring a cosmetic change and then dismiss all discussion of that change as bikeshedding.

As the person raising the change you should provide documentation and standards for nixpkgs before demanding them from others.

Reply:
My concern is that there is
* **little to no documentation** other than looking at other PRs
* on other PRs committers ask for the exact opposite change
* as mentioned before, "common" does not matter, because there is obviously a lot of bad practices across `nixpkgs` and yet we do not dismiss improvements (and this sentence, paraphrased, has been told to contributors on other PRs, in a far more condescending manner, as I have observed.)

Reply:
Also, I just cited the exact same link and reverted the change already, as mentioned above.